### PR TITLE
Polling Transaction Status

### PR DESCRIPTION
**Pull request by viv-defy**

### CODE REVIEW

File: bundler/src/db/dao/transaction_dao.rs [hunk 1]
```diff
@@ -202,6 +270,7 @@ impl UserTransaction {
     }
 }
 
+#[derive(Default, Clone)]
```
Comment (vishnukvnvv) [inline bundler/src/db/dao/transaction_dao.rs:273]:
```suggestion
#[derive(Default)]
```

**Review by vishnukvnvv — CHANGES_REQUESTED** [review]

File: bundler/src/db/dao/transaction_dao.rs [hunk 2]
```diff
@@ -101,6 +102,73 @@ impl TransactionDao {
...
+            "SELECT t1.*, t1.type as transaction_type, t2.exponent from user_transactions t1 left \
+            join supported_currencies t2 on t1.currency = t2.currency \
+            where transaction_id = $1 and user_address = $2",
+            txn_id,
+            user_wallet_address,
+        );
+        let result = query.fetch_one(pool).await;
+        return match result {
+            Ok(row) => {
+                let mut metadata: TransactionMetadata = Default::default();
```
Comment (vishnukvnvv) [inline bundler/src/db/dao/transaction_dao.rs:121]:
```suggestion
                let mut metadata = TransactionMetadata::default();
```

Comment (vishnukvnvv) [inline bundler/src/db/dao/transaction_dao.rs:121]:
```suggestion
                let mut metadata = TransactionMetadata::default();
```

File: bundler/src/db/dao/transaction_dao.rs [hunk 3]
```diff
@@ -101,6 +102,73 @@ impl TransactionDao {
...
+                        metadata,
+                        created_at: row.created_at.to_string(),
+                        updated_at: row.updated_at.to_string(),
+                    },
+                    exponent: row.exponent,
+                }
+            }
+            Err(error) => {
+                error!("Failed to fetch transactions: {:?}", error);
+                Default::default()
```
Comment (vishnukvnvv) [inline bundler/src/db/dao/transaction_dao.rs:151]:
```suggestion
                UserTransactionWithExponent::default()
```

File: bundler/src/handlers/wallet.rs [hunk 4]
```diff
@@ -71,3 +74,24 @@ pub async fn list_transactions(
...
+    }
+
+    let transaction =
+        TransferService::get_status(db_pool.get_ref(), query.transaction_id.clone(), user_id)
+            .await
+            .unwrap();
+
+    Ok(HttpResponse::Ok().json(BaseResponse {
+        data: transaction,
+        err: Default::default(),
```
Comment (vishnukvnvv) [inline bundler/src/handlers/wallet.rs:95]:
```suggestion
        err: Error::default(),
```

**Review by vishnukvnvv — CHANGES_REQUESTED** [review]

File: bundler/src/handlers/wallet.rs (hunk 4 above)
Comment (ska1296) [reply bundler/src/handlers/wallet.rs:95]: @vishnukvnvv why?

**Review by ska1296 — CHANGES_REQUESTED** [review]

**Review by vishnukvnvv — APPROVED** [review]

**Pull request merged**